internal/pkg/service/common/etcdop/watch_test.go

```diff
@@ -14,45 +15,76 @@ import (
 func TestTypedPrefix_Watch(t *testing.T) {
 	t.Parallel()
 
-	ctx := context.Background()
-	c := etcdhelper.ClientForTest(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
+	c := etcdhelper.ClientForTest(t)
 	pfx := typedPrefixForTest()
-	ch := pfx.Watch(ctx, c, func(err error) {
-		assert.FailNow(t, err.Error())
-	})
+
+	// Create watcher
+	errHandler := func(err error) {
+		assert.FailNow(t, `unexpected watch error`, err.Error())
+	}
+	ch := pfx.Watch(ctx, c, errHandler, etcd.WithRev(0))
+
+	// PUT key
 	go func() {
-		key1 := pfx.Key("key1")
-		err := key1.Put("foo").Do(ctx, c)
-		assert.NoError(t, err)
+		assert.NoError(t, pfx.Key("key1").Put("foo").Do(ctx, c))
 	}()
-	msg := <-ch
-	fooVal := fooType("foo")
-	assert.Equal(t, EventT[fooType]{
-		Value: &fooVal,
-		Event: &etcd.Event{
-			Type: mvccpb.PUT,
-			Kv: &mvccpb.KeyValue{
-				Key:   []byte("my/prefix/key1"),
-				Value: []byte(`"foo"`),
+
+	// Wait for PUT event
+	putDone := make(chan struct{})
+	go func() {
+		fooVal := fooType("foo")
+		assert.Equal(t, EventT[fooType]{
+			Value: &fooVal,
+			Event: &etcd.Event{
+				Type: mvccpb.PUT,
+				Kv: &mvccpb.KeyValue{
+					Key:   []byte("my/prefix/key1"),
+					Value: []byte(`"foo"`),
+				},
 			},
-		},
-	}, clearEvent(msg))
+		}, clearEvent(<-ch))
+		close(putDone)
+	}()
+	assert.Eventually(t, func() bool {
```

Comment:
Otestovat sa to da napr. tak, ze sa budeme kazdy `10ms` pytat, ci sa TO uz stalo.
Maximalne ale `5s`.